DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/README.md

```diff
@@ -0,0 +1,354 @@
+
+## Key Implementation Details
+This SVD design consists of 4 computation kernels, as well as several memory access kernels to handle input and output. These kernels are connected through inter-kernel pipes and input/output through unified shared memory (USM).
+
+![](assets/SVD.svg)
+
+### Source file structure
+| File name              | Type                            | Content                                                                                                                                                                                              |
+| ---------------------- | ------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| svd_demo.cpp           | Host code                       | Launch a demonstration using the SVD design. Contains the `main()` function.                                                                                                                         |
+| svd_testcase.hpp       | Host code                       | A `struct` that lunches the design with a set of input and check result for correctness. The input can either be specified or generated.                                                             |
+| svd_testbench_tool.hpp  | Host code                       | Helper functions that are used in the test bench of the SVD demonstration to check for correctness.                                                                                                   |
+| print_matrix.hpp       | Host code                       | Helper functions to print matrices.                                                                                                                                                                  |
+| golden_pca.hpp         | Host code                       | A CPU reference PCA design that is used to calculate reference eigenvalues for the test bench.                                                                                                      |
+| svd.hpp                | Host code with some device code | Contains wrapper function `SingularValueDecomposition` that launches individual kernels of the SVD design.                                                                                           |
+| memory_transfers.hpp   | Device code                     | Contains kernel implementations of `MatrixReadFromDDRTo2PipesByBlocks` , `MatrixReadPipeToDDR` and `VectorReadPipeToDDR`. These kernels transfers data between DDR and streaming interfaces (pipes).|
+| usv_from_eigens.hpp       | Device code                     | Contains kernel implementation of `USVFromEigens`.                                                                                                                                                   |
```

Comment:
You should add references to the shared files in the `include` directory to this table, but make it clear that those files are shared.